src/frontend/frontend/file.js

```diff
@@ -384,7 +380,7 @@ angular.module('frontend-app')
         self.runFile = function() {runWhenSaved(function () {
           self.killProgram().then(function() {
             self.console.clear();
-            self.project.run(self.question, "question", self.runnerFile, self.contents, false)
+            self.project.run(self.question, "question", self.file, self.contents, false)
```

Comment:
This is confusing, as self.file is not really used (outside of the first run).  We should remove this parameter entirely.